### DISCUSSION TITLE
Concerns about minimum plot size of k=32

### DISCUSSION THREAD

**Discussion by mjmandl** [Chia-Network/chia-blockchain#597] [Q&A] [answered]
While I understand that the idea of someone possibly being able to re-plot (phase 1) in under 5 minutes could be an issue I would like to have a better understanding of whether or not this is a realistic scenario. Since plotting does use resources would it really be worth it to plot a k=31 plot between every block (just to get a relatively small increase in effective space)?

It seems to be that the k=32 minimum plot size might be prohibitive to smaller farmers (some people have 50-150gb free on their pc and not enough space for a 300gb temp file). And if the goal is to be as decentralized as possible it seems to me that these smaller farmers should be encouraged.

Additionally, allowing smaller plot sizes would make it easier to plot on an SSD with limited space and then move the file to your hard drive. In my case I have 100gb free on my SSD but can unfortunately not utilize this to plot the 300gb temp files. This means I have to plot directly onto my 8tb hard drive which further increases the time needed to plot everything.

I am relatively new so it is very possible that I have overlooked something, maybe some of you could give me some more insight. Thanks in advance! :)

**Comment by hoffmang9** [comment, marked as answer]
There are two additional issues that have to be kept in mind.

The first is that the single fastest way to plot through only phase 1 of a plot - as one can theoretically farm just phase 1 - is to use a very large memory machine and perform all the sorts in memory. That's mostly limited by the CPU to RAM speed actually and it is MUCH faster than you'd expect already today. It's not cost effective and has a very low per day TB of plots written, but it's an attack vector that can turn Proof of Space back into Proof of Work. You can create lots more farmable TB by plotting slower and as parallel as you can. The plotting and farming process is designed to be very parallelize-able. 

The second is that all k sizes will eventually go up due to a mix of Moore's and [Kryder's](https://www.scientificamerican.com/article/kryders-law/) laws. A k=31 would be likely retired much sooner than a k=32. At k=32 and knowing what we do about RAM and disk prices and technologies today - we think that k=32 will make it certainly 7 years and probably 10+ years.

There are always tradeoffs but we think erring on the side of lowering the amount of re-plotting necessary and keeping our security constants pretty high is the right trade.

-Gene

**Comment by lexick2016** [comment]
Can someone explain to me aft the plots are going into the temp ssd/space then it moves on the Permanent hdd (in my case my nas).
How long it stays there?
How much space you need on the permanent hdds?
What happens when the hdds(permanent ones) are full? You stop making money? You are still making money? Are you still plotting on the ssd?
I would really appreciate your help to understand this. Thank you

**Reply by dorkmo** [reply]
after you move to permanent space, the plots stay there and are continuously used for "farming". once you fill up your desired hdd's you cant stop plotting and simply farm.

**Reply by Lykos153** [reply]
Chiming in to state the probably obvious, but as the typo literally negates the meaning of the answer to the very question that was posed, I think it's worth pointing out that what @dorkmo wanted to say was:

Once you fill up your desired HDDs you **can** stop plotting and simply farm.

**Reply by Chiazz** [reply]
WHen my plot are moving to my HDD after 100% ploting, the size is just about 4.2GO.........i really do'nt understand why, somebody have an idea?

**Reply by 13Trades** [reply]
Check your block size.

**Reply by Andrey5045** [reply]
Everybody suggests to plot k32 instead of k34, I did not follow it because I thought that I would look a little bit ahead as k32 will go in a few years. Seven? C'mon, when bitcoin was created they said that the last coin would be mined sometime in 2100, but they counted it without taking into consideration that GPUs will be used, then GPU rigs, pools and finally ... ASICs. Number of bitcoins mined is 90%. So I think that k32 will go much quicker. Who knows what technology brings? I predict that one day home computers' RAM will be in TBs and HDDs will become virtual in RAM. Proof of Space&Time might become Proof of Work again, as @hoffmang9 says.
Now I will tell you about my experience with k34:
I tried to plot k34 with HDD. Computing Table3 @Phase1 took 5 days to complete, and according to calculated time Table4 @Phase 1 would tool 19 days!! I thought that it was slow because it was HDD not SDD. I got an external SDD (I got it specially to kill it by plotting as I did not want to mess with my primary internal SDD) and guess what.... SAME TIME! (6 days to reach 7%)
Probably my computer is too old for k34 - Intel i7-4770K@3.5GHz with 8GB RAM.
I stopped plotting k34 and started to plot k32: on SDD (not on HDD), results are much more promising: 72% in 18 hours, it is on Phase3 of 4 working with tables 4&5.

I am not patient to plot one k34 in one month, I better plot 10 k32s in 10 days and start farming. But surely future holds larger k-sizes. Just like with bitcoin, the entry level is not accessible to a simple home computer, Chia might become same type of cryptocurrency, too.

**Reply by hoffmang9** [reply]
The obsolescence of k32 in less than a decade would require a fundamental - likely discovery in physics - change in the current CPU and RAM roadmaps from all the major players. 

This isn't like Bitcoin in that way. The ultimate hardware methods are generally understood and the economics strongly force people to just plot k32s instead of trying to grind them.
